contracts/contracts/UserReplicaSetManager.sol

```diff
@@ -0,0 +1,416 @@
+pragma solidity ^0.5.0;
+
+import "./interface/RegistryInterface.sol";
+import "./registry/RegistryContract.sol";
+import "./UserFactory.sol";
+
+import "./SigningLogicInitializable.sol";
+
+
+/** @title Contract for Audius user replica set management */
+contract UserReplicaSetManager is SigningLogicInitializable, RegistryContract {
+    // Reference to data contract registry
+    RegistryInterface private registry = RegistryInterface(0);
+    // User factory key used to confirm valid users during reconfig operations
+    bytes32 private userFactoryRegistryKey;
+
+    // Address permissioned to update user replica sets
+    address userReplicaSetBootstrapAddress;
+
+    // spID to ServiceProvider delegateOwnerWallet
+    mapping (uint => address) spIdToContentNodeDelegateWallet;
+
+    // Struct used to represent replica sets
+    // Each uint reprsets the spID registered on eth-contracts
+    struct ReplicaSet {
+        uint primary;
+        uint[] secondaries;
+    }
+
+    // Consider frontrunning of spIDs
```

Comment:
since this is now populated in the constructor, do we still need to store the fields on chain for any reason?

Reply:
initially had kept it for a record of the 'initial' nodes vs later nodes...but now there isn't really any speecial permission for these nodes since we require 3 cnode auth to add a new node later on. debated an event but idt we can have an event in the constructor itself